Frontend/HsTypes.hs

```diff
@@ -119,7 +119,17 @@ data Term a = TmVar (HsTmVar a)                   -- ^ Term variable
 type PsTerm = Term Sym
 type RnTerm = Term Name
 
-data HsAlt a = HsAlt (HsPat a) (Term a)
+data HsGuard   a = HsGuard (HsPat a) (Term a)
```

Comment:
Maybe rename the data constructor to `HsPatGuard`? We would like to (eventually) have boolean guards as well